File: ecmascript/trinary/trinary.js
export default class Trinary {
  constructor(n){
    this.n = n;
  }
  isTrinary(n){
    let re = /^[012]+$/;
    return re.test(n);
  }
  toDecimal(){
    if (!this.isTrinary(this.n)) { return 0; }

    return [...this.n].reverse().reduce((o,v,i) => {
      return o + Number(v) * Math.pow(3,i);
    }, 0);
  }
}

File: ruby/trinary/trinary.rb
class Trinary
  def initialize(num)
    @num = num.gsub("\n", "")
  end

  def to_decimal
    return 0 if (@num =~ /^[0-2]+$/).nil?
    @num.reverse.chars.map.with_index {|n, i| n.to_i * 3 ** i }.reduce(:+)
  end

end

module BookKeeping
  VERSION = 1
end
